Sources/XcodeGenKit/SettingsBuilder.swift

```diff
@@ -23,6 +23,13 @@ extension ProjectSpec {
             buildSettings += SettingsPresetFile.config(type).getBuildSettings()
         }
 
+        if let configPath = configFiles[config.name] {
+            // Do not overwrite project xcconfig's value.
+            if let configFile = try? XCConfig(path: basePath + configPath) {
```

Comment:
Can we move these config file lines into the common function as well?
```
if let configPath = configFiles[config.name] {
   buildSettings = removeConfigFileSettings(from: buildSettings, configPath: configPath)
}
```

That way it's all kept together, removing duplication. It then also lets us potentially cache the loaded and flattened file settings in one place, because if you have many targets each one of them could be loading and flattening a LOT of files

Reply:
Should I let you refactor those after merge? I think you have solid image.
I wouldn't mind about my code being modified at all.😉